File: Rh.Entities/RhEntrevista/VagaTecnologia.cs
using System.Collections.Generic;

namespace Rh.Entities.RhEntrevista
{
    public class VagaTecnologia
    {
        public VagaTecnologia()
        {
            ListaEntrevistaTecnologia = new HashSet<EntrevistaTecnologia>();
        }

        public int VagaTecnologiaId { get; set; }
        public int VagaId { get; set; }
        public int TecnologiaId { get; set; }
        public int? Peso { get; set; }

        public virtual Vaga  Vaga { get; set; }
        public virtual Tecnologia Tecnologia { get; set; }

        public ICollection<EntrevistaTecnologia> ListaEntrevistaTecnologia { get; set; }
    }
}

File: Rh.Application/ClientApp/app/_models/vagaTecnologia.ts
export interface VagaTecnologia {
    vagaTecnologiaId: number;
    vagaId: number;
    tecnologiaId: number;
    peso?: number;
    selected: boolean;
}

export interface VagaTecnologiaEntrevista {
    vagaTecnologiaId: number;
    tecnologiaNome: string;
}